vortex-ffi/src/error.rs

```diff
@@ -0,0 +1,52 @@
+use std::ffi::{c_char, c_int};
+use std::ptr;
+
+use vortex::error::VortexResult;
+
+#[repr(C)]
+pub struct FFIError {
+    pub code: c_int,
+    pub message: *const c_char,
+}
+
+pub unsafe fn into_c_error_id<V>(
```

Comment:
i think error_id is a bit of a confusing name (i at first thought it was something about error codes).

what are your thoughts on

- into_c_error -> this function
- map_into_c_error -> the variant below that takes a mapping argument for the result

or just make this one macro with two entrypoints?

Reply:
good idea